src/main/java/org/spongepowered/common/data/processor/data/entity/PassengerDataProcessor.java

```diff
@@ -0,0 +1,207 @@
+            final Entity newVehicle = manipulator.vehicle().get();
+
+            try {
+                builder.replace(
+                        new ImmutableSpongeValue<Entity>(Keys.VEHICLE, prevVehicle),
+                        new ImmutableSpongeValue<Entity>(Keys.BASE_VEHICLE, prevBaseVehicle));
+                entity.ridingEntity = (net.minecraft.entity.Entity) newVehicle;
+                builder.success(
+                        new ImmutableSpongeValue<Entity>(Keys.BASE_VEHICLE, newBaseVehicle),
+                        new ImmutableSpongeValue<Entity>(Keys.VEHICLE, newVehicle))
+                        .result(DataTransactionResult.Type.SUCCESS);
+                return builder.build();
+            } catch (Exception e) {
+                entity.riddenByEntity = (net.minecraft.entity.Entity) prevVehicle;
+                builder.reject(
+                        new ImmutableSpongeValue<Entity>(Keys.VEHICLE, newVehicle),
+                        new ImmutableSpongeValue<Entity>(Keys.BASE_VEHICLE, newBaseVehicle))
+                        .result(DataTransactionResult.Type.ERROR);
+                return builder.build();
+            }
+        }
+        return DataTransactionBuilder.failResult(manipulator.getValues());
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, PassengerData manipulator, MergeFunction function) {
+        return null;
+    }
+
+    @Override
+    public Optional<ImmutablePassengerData> with(Key<? extends BaseValue<?>> key, Object value, ImmutablePassengerData immutable) {
+        return Optional.absent();
+    }
+
+    @Override
+    public DataTransactionResult remove(DataHolder dataHolder) {
+        return DataTransactionBuilder.builder().result(DataTransactionResult.Type.FAILURE).build();
+    }
+
+    @Override
+    public PassengerData create() {
+        return new SpongePassengerData();
+    }
+
+    @Override
+    public ImmutablePassengerData createImmutable() {
+        return new ImmutableSpongePassengerData(null, null);
+    }
+
+    @Override
+    public Optional<PassengerData> createFrom(DataHolder dataHolder) {
+        if (dataHolder instanceof net.minecraft.entity.Entity) {
+            final Entity vehicle = (Entity) ((net.minecraft.entity.Entity) dataHolder).riddenByEntity;
+
+            net.minecraft.entity.Entity currentEntity = ((net.minecraft.entity.Entity) dataHolder).ridingEntity;
+
+            while (currentEntity != null && currentEntity.isRiding())
+            {
+                currentEntity = currentEntity.ridingEntity;
+            }
+
+            final Entity baseVehicle = (Entity) currentEntity;
+
+            return Optional.<PassengerData>of(new SpongePassengerData(vehicle, baseVehicle));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<PassengerData> build(DataView container) throws InvalidDataException {
+        return Optional.absent();
```

Comment:
This is the exact same as deserializing, it shouldn't be `absent`, rather, it can't be `absent` for the manipulator to work.